PolyEngine/Core/Src/EnumUtils.hpp

```diff
@@ -126,18 +128,22 @@ namespace Poly {
 
 	namespace Impl {
 		template<typename T>
-		struct EnumInfo {
-		};
+		struct EnumInfo {};
 	}
 
 	template<typename T>
 	const char* GetEnumName(T val)
 	{
-		return Poly::Impl::EnumInfo<T>::Get().Names[val];
+		return Impl::EnumInfo<T>::Get().Names[val];
 	}
 }
 
-
-#define REGISTER_ENUM_NAMES(type, ...) template<> struct Poly::Impl::EnumInfo<type> { \
-	static Poly::Impl::EnumInfo<type>& Get() {static Poly::Impl::EnumInfo<type> instance; return instance; } \
-	const Poly::EnumArray<const char*, type> Names = {__VA_ARGS__}; }
\ No newline at end of file
+//NOTE(vuko): apparently defining specializations in a namespace from global/other namespace is illegal C++ and GCC complains
+//Unfortunately being compliant causes problems when using the macro in a namespace. If you run into problems, close the namespace, use the macro, then open it again
+#define REGISTER_ENUM_NAMES(type, ...)                                                    \
+	namespace Poly { namespace Impl {                                                     \
```

Comment:
Yes, I spent a lot of time on this trying to get it work without workarounds... Right now I close the namespace, use the macro, then open it again. It works, but two macros seems neater.